### PR TITLE
Add LZO 2.09 recipe.

### PR DESCRIPTION
**Pull request by rolando**
This recipe builds shared libraries along with static libraries.

- [x] Linux build
- [x] OSX build
- [x] Windows build
- [ ] Ensure compatibility with existing feedstocks

### CODE REVIEW

**Comment by rolando** [comment]
@conda-forge/core Windows builds are having an error that seems unrelated to the recipe itself:

```
...Traceback (most recent call last):
  File "C:\Miniconda\Scripts\conda-build-all-script.py", line 5, in <module>
    sys.exit(conda_build_all.cli.main())
  File "C:\Miniconda\lib\site-packages\conda_build_all\cli.py", line 90, in main
    b.main()
  File "C:\Miniconda\lib\site-packages\conda_build_all\builder.py", line 267, in main
    built_dist_location = self.build(meta, build_config)
  File "C:\Miniconda\lib\site-packages\conda_build_all\builder.py", line 207, in build
    with meta.vn_context():
  File "C:\Miniconda\lib\site-packages\conda_build_all\resolved_distribution.py", line 40, in vn_context
    return setup_vn_mtx_case(self.special_versions, config)
  File "C:\Miniconda\lib\site-packages\conda_build_all\version_matrix.py", line 341, in setup_vn_mtx_case
    config.CONDA_PY = version
AttributeError: 'NoneType' object has no attribute 'CONDA_PY'
```

I need assistance regarding the windows options. I'm not sure if the `build/features` section is right and if the build script and test section for `win` is OK. I have followed other recipes as examples (i.e.: `zlib`).

**Comment by rolando** [comment]
Appveyor is replying with
```
Cannot read last commit information from GitHub repository conda-forge/staged-recipes: HTTP GET to https://api.github.com/repos/conda-forge/staged-recipes/branches/master returned status code 403: Forbidden Full response body: {"message":"API rate limit exceeded for conda-forge-admin.","documentation_url":"https://developer.github.com/v3/#rate-limiting"}
```

**Comment by jakirkham** [comment]
xref: https://github.com/conda-forge/conda-forge.github.io/issues/314

Though it looks like it built ok FWICT.

**Comment by rolando** [comment]
Re-pushed and last build is OK :tada:

**Comment by scopatz** [comment]
LGTM Thanks @rolando

File: recipes/lzo/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: http://www.oberhumer.com/opensource/{{ name }}/download/{{ name }}-{{ version }}.tar.gz
+  {{ hash_type }}: {{ hash_val }}
+  patches:
+    # Patch imported from https://github.com/Microsoft/vcpkg/tree/ed8e7ca79dad5b7bcf716efe25d975571e8e8c05/ports/lzo
+    - do-not-declare-setargv.patch  # [win]
```
Comment (jakirkham) [inline recipes/lzo/meta.yaml]: Has this already been raised upstream. Maybe they can add this within an `#ifdef` block.

Comment (rolando) [inline recipes/lzo/meta.yaml]: As far I can see in the [homepage](http://www.oberhumer.com/opensource/lzo/), there is no public issue tracker or development repository where we can report this issue.

Comment (jakirkham) [inline recipes/lzo/meta.yaml]: 😢 Yep, looks like you're right.

Though I think I did find his email. Can send it to you.

Also there appears to be a GitHub mirror ( https://github.com/nemequ/lzo ).

Comment (rolando) [inline recipes/lzo/meta.yaml]: Nice find. Although commits are just releases imports :/ https://github.com/nemequ/lzo/commits/master

Comment (rolando) [inline recipes/lzo/meta.yaml]: Giving it a try: https://github.com/nemequ/lzo/issues/1

Comment (jakirkham) [inline recipes/lzo/meta.yaml]: > commits are just releases imports

Yeah maybe I should have said "mirror". Though they appear to attribute the right author.

> Giving it a try: https://github.com/nemequ/lzo/issues/1

Thanks @rolando. Bare minimum they can tell us where to send the patch.

**Pull request merged**